internal/build/phases.go

```diff
@@ -280,6 +309,32 @@ func (l *Lifecycle) newExport(repoName, runImage string, publish bool, launchCac
 	return phaseFactory.New(configProvider), nil
 }
 
+func (l *Lifecycle) writeStackToml() (string, error) {
+	buf := &bytes.Buffer{}
+	err := toml.NewEncoder(buf).Encode(l.builder.Stack())
+	if err != nil {
+		return "", errors.Wrap(err, "marshaling stack metadata")
+	}
+
+	var stackFile *os.File
+	if stackFile, err = ioutil.TempFile("", "stack.toml"); err != nil {
+		return "", errors.Wrap(err, "opening stack.toml tempfile")
+	}
+
+	if _, err = stackFile.Write(buf.Bytes()); err != nil {
+		return "", errors.Wrapf(err, "writing stack.toml tempfile: %s", stackFile.Name())
+	}
+
+	// Override umask
```

Comment:
umask (needed for Linux) and symlinks (needed for Mac) requirements were uncovered in our acceptance tests for volume mounts.